### ISSUE TITLE
Standard Conflict: JumpStatementsSpacing & BlockControlStructureSpacing

### ISSUE THREAD

**Issue by yob-yob** [slevomat/coding-standard#1276]
Code

```php
private function generateThumbnail()
{
    switch ($this->record->type) {
        case 'application/pdf':
            return asset('thumbnails/pdf.thumb.png');

        default:
            return asset('thumbnails/video.thumb.png');
    }
}
```

phpcs config file

```xml
<?xml version="1.0"?>
<ruleset name="Custom CS">
  <!-- Use Slevomat Coding Standard -->
  <config name="installed_paths" value="../../slevomat/coding-standard"/>

  <!-- Phpcs Arguments -->
  <arg name="basepath" value="."/>
  <!-- Show progress of the run -->
  <arg value="p"/>
  <!-- Show sniff codes in all reports -->
  <arg value="s"/>
  <!-- Explain a standard by showing the sniffs it includes -->
  <!-- <arg value="e"/> -->
  <!-- Use colors in output -->
  <arg name="colors"/>

  <!-- Files -->
  <file>./app</file>
  <file>./config</file>
  <file>./database</file>
  <file>./resources</file>
  <file>./routes</file>
  <file>./tests</file>

  <!-- Global Excludes -->
  <exclude-pattern>*.blade.php</exclude-pattern>

  <!-- Rules -->

  <!-- Formating -->
    <rule ref="SlevomatCodingStandard.Arrays"/>
    <rule ref="SlevomatCodingStandard.Classes.ClassMemberSpacing"/>
    <rule ref="SlevomatCodingStandard.Classes.ConstantSpacing"/>
    <rule ref="SlevomatCodingStandard.Classes.DisallowMultiConstantDefinition"/>
    <rule ref="SlevomatCodingStandard.Classes.DisallowMultiPropertyDefinition"/>
    <rule ref="SlevomatCodingStandard.Classes.MethodSpacing"/>
    <rule ref="SlevomatCodingStandard.Classes.ModernClassNameReference"/>
    <rule ref="SlevomatCodingStandard.Classes.ParentCallSpacing"/>
    <rule ref="SlevomatCodingStandard.Classes.PropertySpacing"/>
    <rule ref="SlevomatCodingStandard.Classes.RequireMultiLineMethodSignature"/>
    <rule ref="SlevomatCodingStandard.Classes.RequireSingleLineMethodSignature"/>
    <rule ref="SlevomatCodingStandard.Classes.SuperfluousAbstractClassNaming"/>
    <rule ref="SlevomatCodingStandard.Classes.SuperfluousInterfaceNaming"/>
    <rule ref="SlevomatCodingStandard.Classes.SuperfluousExceptionNaming"/>
    <rule ref="SlevomatCodingStandard.Classes.SuperfluousErrorNaming"/>
    <rule ref="SlevomatCodingStandard.Classes.SuperfluousTraitNaming"/>
    <rule ref="SlevomatCodingStandard.Classes.TraitUseDeclaration"/>
    
    <rule ref="SlevomatCodingStandard.Classes.TraitUseSpacing">
      <!-- Fix Conflicts with -->
        <!-- PSR12.Traits.UseDeclaration.UseAfterBrace -->
        <!-- PSR12.Traits.UseDeclaration.BlankLineAfterLastUse -->
        <!-- PSR2.Classes.ClassDeclaration.CloseBraceAfterBody -->
      <properties>
        <property name="linesCountAfterLastUseWhenLastInClass" value="0" />
        <property name="linesCountBeforeFirstUseWhenFirstInClass" value="0" />
      </properties>
    </rule>

    <rule ref="SlevomatCodingStandard.ControlStructures.BlockControlStructureSpacing" />
    
    <rule ref="SlevomatCodingStandard.ControlStructures.JumpStatementsSpacing"/>
</ruleset>
```

based on the code... this config will cause an error

```
Expected 0 lines after "return", found 1.
(SlevomatCodingStandard.ControlStructures.JumpStatementsSpacing.IncorrectLinesCountAfterLastControlStructure)
```

to fix this... code will be edited to 

```php
private function generateThumbnail()
{
    switch ($this->record->type) {
        case 'application/pdf':
            return asset('thumbnails/pdf.thumb.png');
        default:
            return asset('thumbnails/video.thumb.png');
    }
}
```

but I still get an error from 

```
Expected 1 line after "case", found 0.
(SlevomatCodingStandard.ControlStructures.BlockControlStructureSpacing.IncorrectLinesCountAfterControlStructure)

Expected 1 line before "default", found 0.
(SlevomatCodingStandard.ControlStructures.BlockControlStructureSpacing.IncorrectLinesCountBeforeControlStructure)
```

I know that I can just fix this by setting this on my xml config

```
<rule ref="SlevomatCodingStandard.ControlStructures.JumpStatementsSpacing">
      <properties>
        <property name="linesCountAfterWhenLastInCaseOrDefault" value="1" />
        <property name="linesCountAfterWhenLastInLastCaseOrDefault" value="0" />
      </properties>
</rule>
```

but I do think this should be document if not fixed ? I only reported this since I find it weird that a standard is having conflict with it's own default pre-configured standards....

**Comment by kukulich** [comment]
It's absolutely right :)

**Comment by yob-yob** [comment]
I'm sorry what does that mean? can you explain a little bit more please... I'm trying to understand my mistake...

**Comment by kukulich** [comment]
@yob-yob It's right that you have to configure both sniffs to be happy with the result. Each sniff works independently.
